osu.Game/Skinning/RulesetSkinProvidingContainer.cs

```diff
@@ -75,6 +83,8 @@ protected virtual void UpdateSkins()
                         break;
                 }
             }
+
+            SkinSources.Add(new RulesetResourcesSkin(Ruleset, host, audio));
```

Comment:
One thing to note here is that you are adding this at the end, meaning the ruleset has no chance to override the samples returned by the default skin implementation. I'm not sure that's what we want - imagine you were making a ruleset that shares the standard sample naming structure as the "official" osu! rulesets but wanted to provide a custom sample set to replace the defaults.

Reply:
Although that's how it was previously when the fallback logic was in `PoolableSkinnableSample` etc., it's a pretty good point. Should be resolved in a simple way now.